counter/src/test/java/org/infinispan/counter/AbstractCounterNotificationTest.java

```diff
@@ -169,32 +171,41 @@ private void printQueue(ListenerQueue queue) {
 
    static class ListenerQueue implements CounterListener {
 
+      private static final Log log = LogFactory.getLog(ListenerQueue.class);
       final BlockingQueue<CounterEvent> queue;
+      final String name;
 
-      ListenerQueue() {
+      ListenerQueue(String name) {
          queue = new LinkedBlockingQueue<>();
+         this.name = name;
       }
 
       @Override
       public void onUpdate(CounterEvent event) {
-         queue.offer(event);
+         try {
+            queue.put(event);
+            log.tracef("[%s] add event %s", name, event);
```

Comment:
isTraceEnabled is not necessary here ?

Reply:
not really, because it is a test. but I'll add it.

Reply:
ah I didn't realise the name of the class!